common/beerocks/bwl/dummy/mon_wlan_hal_dummy.cpp

```diff
@@ -95,15 +150,198 @@ bool mon_wlan_hal_dummy::sta_link_measurements_11k_request(const std::string &st
+
+        SStaStats dummy_sta_stats = {};
+
+        // Format rssi= %d,%d,%d,%d
+        auto samples = beerocks::string_utils::str_split(tmp_str, ',');
+        for (auto &s : samples) {
+            float s_float = float(beerocks::string_utils::stoi(s));
+            if (s_float > beerocks::RSSI_MIN) {
+                dummy_sta_stats.rx_rssi_watt += std::pow(10, s_float / float(10));
+                dummy_sta_stats.rx_rssi_watt_samples_cnt++;
+            }
+        }
+
+        if (!dummy_obj_read_str("snr", parsed_obj, &tmp_str)) {
+            LOG(ERROR) << "Failed reading snr!";
+            return false;
+        }
+
+        // Format snr=%d,%d,%d,%d
+        auto samples_snr = beerocks::string_utils::str_split(tmp_str, ',');
+        for (auto &s : samples_snr) {
+            float s_float = float(beerocks::string_utils::stoi(s));
+            if (s_float >= beerocks::SNR_MIN) {
+                dummy_sta_stats.rx_snr_watt += std::pow(10, s_float / float(10));
+                dummy_sta_stats.rx_snr_watt_samples_cnt++;
+            }
+        }
+
+        if (!dummy_obj_read_int("uplink", parsed_obj, tmp_int)) {
+            LOG(ERROR) << "Failed reading rsni parameter!";
+            return false;
+        }
+        dummy_sta_stats.tx_phy_rate_100kb = (tmp_int / 100);
+
+        if (!dummy_obj_read_int("downlink", parsed_obj, tmp_int)) {
+            LOG(ERROR) << "Failed reading rsni parameter!";
+            return false;
+        }
+        dummy_sta_stats.rx_phy_rate_100kb = (tmp_int / 100);
+
+        m_dummy_stas_map[sta_mac].sta_stats = dummy_sta_stats;
+        break;
+    }
+    case Data::RRM_Update_Beacon_Measurements: {
+        if (!dummy_obj_read_str(DUMMY_EVENT_KEYLESS_PARAM_MAC, parsed_obj, &sta_mac)) {
+            LOG(ERROR) << "Failed reading mac parameter!";
+            return false;
+        }
+
+        if (!dummy_obj_read_str("bssid", parsed_obj, &bssid)) {
+            LOG(ERROR) << "Failed reading bssid!";
+            return false;
+        }
+
+        m_dummy_stas_map[sta_mac].beacon_measurment_events[bssid] = parsed_obj;
+        break;
+    }
+    default:
+        LOG(WARNING) << "Unhandled event received: " << opcode;
+        break;
+    }
+
+    return true;
+}
+
 bool mon_wlan_hal_dummy::process_dummy_event(parsed_obj_map_t &parsed_obj)
 {
+    char *tmp_str;
+    int64_t tmp_int;
     // Filter out empty events
     std::string opcode;
     if (!(parsed_obj.find(DUMMY_EVENT_KEYLESS_PARAM_OPCODE) != parsed_obj.end() &&
           !(opcode = parsed_obj[DUMMY_EVENT_KEYLESS_PARAM_OPCODE]).empty())) {
         return true;
     }
     LOG(TRACE) << __func__ << " - opcode: |" << opcode << "|";
+
+    auto event = dummy_to_bwl_event(opcode);
+
+    switch (event) {
+    case Event::RRM_Beacon_Response: {
+        // Allocate response object
+        auto resp_buff = ALLOC_SMART_BUFFER(sizeof(SBeaconResponse11k));
+        auto resp      = reinterpret_cast<SBeaconResponse11k *>(resp_buff.get());
```

Comment:
So this code can move directly into sta_beacon_11k_request().

Reply:
See comment above.